src/mlpack/tests/loss_functions_test.cpp

```diff
@@ -578,4 +579,92 @@ BOOST_AUTO_TEST_CASE(HingeEmbeddingLossTest)
   BOOST_REQUIRE_EQUAL(output.n_rows, input.n_rows);
   BOOST_REQUIRE_EQUAL(output.n_cols, input.n_cols);
 }
+
+/**
+ * Simple test for the Cosine Embedding loss function.
+ */
+BOOST_AUTO_TEST_CASE(CosineEmbeddingLossTest)
+{
+  arma::mat input1, input2, y, output;
+  double loss;
+  CosineEmbeddingLoss<> module;
+
+  // Test the Forward function. Loss should be 0 if input1 = input2 and y = 1.
+  input1 = arma::mat(1, 10);
+  input2 = arma::mat(1, 10);
+  input1.ones();
+  input2.ones();
+  y = arma::mat(1, 1);
+  y.ones();
+  loss = module.Forward(input1, input1);
+  BOOST_REQUIRE_SMALL(loss, 1e-6);
+
+  // Test the Backward function.
+  module.Backward(input1, input1, output);
+  BOOST_REQUIRE_SMALL(arma::accu(output), 1e-6);
+
+  // Check for dissimilarity.
+  module.Similarity() = false;
+  loss = module.Forward(input1, input1);
+  BOOST_REQUIRE_CLOSE(loss, 1.0, 1e-4);
+
+  // Test the Backward function.
+  module.Backward(input1, input1, output);
+  BOOST_REQUIRE_SMALL(arma::accu(output), 1e-6);
+
+  input1 = arma::mat(3, 2);
+  input2 = arma::mat(3, 2);
+  input1.fill(1);
+  input1(4) = 2;
+  input2.fill(1);
+  input2(0) = 2;
+  input2(1) = 2;
+  input2(2) = 2;
+  loss = module.Forward(input1, input2);
+  // Caclulated using torch.nn.CosineEmbeddingLoss().
```

Comment:
```suggestion
  // Calculated using torch.nn.CosineEmbeddingLoss().
```

Reply:
Hey @favre49, Thanks a lot for the review. I will make the changes as soon as possible.

Reply:
Done. :)